sympy/physics/control/lti.py

```diff
@@ -0,0 +1,121 @@
+from sympy import (
+    Basic, Matrix, ShapeError, Mul,
+    ImmutableMatrix, MutableMatrix, MutableDenseMatrix,
+)
+from sympy.polys.polytools import degree
+
+__all__ = ['TransferFunctionMatrix', 'TransferFunction']
+
+_matrixTypes = (
+    Matrix, ImmutableMatrix, MutableMatrix, MutableDenseMatrix)
+
+
+class TransferFunctionMatrix(Basic):
+    def __new__(cls, args):
+        # constructor is from a given transfer function.
+        if  isinstance(args, (Matrix, ImmutableMatrix, MutableMatrix)):
+            G = args
+            obj = Basic.__new__(cls, G)
+            obj.G = args
```

Comment:
Don't set attributes like this in Basic subclasses. You can make a property called `G` that returns `self.args[0]`.